### PR TITLE
Spark 3.5: Adapt to Spark 3.5.4

### PR DESCRIPTION
**Pull request by pan3793**
Adapt to Spark 3.5.4, see more details at SPARK-50235, SPARK-50463

### CODE REVIEW

**Review by LuciferYang — APPROVED** [review]

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java [hunk 1]
```diff
@@ -59,6 +59,10 @@ public void close() {
     accessor.close();
   }
 
+  // If a column vector is writable or constant, it should override this method and do nothing.
+  // See more details at SPARK-50235, SPARK-50463 (Fixed in Spark 3.5.4)
+  public void closeIfFreeable() {}
```
Comment (nastra) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: this should probably have an `@Override`and the comment can go inside the method

Comment (LuciferYang) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: No, we cannot use `@Override` until we upgrade to Spark 3.5.4.

Comment (nastra) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: right, but I don't think there's any value in adding this method without actually being on Spark 3.5.4. Does that mean you're also upgrading to 3.5.4 as part of this PR (once it's out)?

Comment (LuciferYang) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: https://github.com/apache/iceberg/pull/11731#issuecomment-2548667713

image

Comment (pan3793) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: w/ or w/o `@Override` won't cause any functional difference.

do you guys want to wait for Spark 3.5.4 to pass the vote, or release Iceberg 1.7.2 ASAP?

Comment (pan3793) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java:64]: >  I don't think there's any value in adding this method without actually being on Spark 3.5.4.

@nastra To not surprise users, I suppose `iceberg-spark-runtime-3.3_2.12-1.7.2.jar` should work for Spark 3.5.x(unfortunately, it's not true for Spark 3.5.3 due to another issue), that means, no matter which version it is built against, it's binary compatible with a serial of versions of Spark 3.5.

In short, the value of this change is, to make the output `iceberg-spark-runtime-3.3_2.12-<v>.jar` built against Spark 3.5.2, to be compatible with Spark 3.5.4

Comment (nastra) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java]: we should probably wait until 3.5.4 is officially out
